### PR TITLE
Feature homekitchen tests

### PR DESCRIPTION
**Pull request by Valkrys**

### CODE REVIEW

**Comment by b-smyth** [comment]
Some files deleted here Jana. If this is merged some peoples work today will be gone.

File: src/main/java/com/sampleweb/pages/HomeKitchen.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
+package com.sampleweb.pages;
+
+import com.sun.imageio.plugins.wbmp.WBMPImageReader;
```
Comment (lewisbirks) [inline src/main/java/com/sampleweb/pages/HomeKitchen.java:3]: Unused import, can delete

File: src/main/java/com/sampleweb/pages/HomeKitchen.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    public String title = "Home & Kitchen";
+    public String secondMenuTitle = "Home & Garden";
+    public static String PATH = "https://www.amazon.co.uk/s/ref=nb_sb_noss?url=search-alias%3Dkitchen&field-keywords=";
+    private String globalStoreBoxLocator = "//input[@name=\"s-ref-checkbox-11259240031\"]";
+    private String globalStoreLabelLocator = "//img[@class=\"s-ref-img-sprite\"]";
+    private String averageCustomerReviewLocator ="//*[@id=\"leftNav\"]/ul[9]";
+    public String secondMenuLocator = "//*[@id=\"nav-subnav\"]/a[1]/span";
+    private String primeProductListLocator = "//*[@class=\"s-result-list s-search-results sg-row\"]//*[@class=\"s-include-content-margin s-border-bottom\"]";
+    private String reviewProductListLocator = "//div[@class=\"sg-col-20-of-24 s-result-item sg-col-0-of-12 sg-col-28-of-32 sg-col-16-of-20 sg-col sg-col-32-of-36 sg-col-12-of-16 sg-col-24-of-28\"]";
```
Comment (lewisbirks) [inline src/main/java/com/sampleweb/pages/HomeKitchen.java:21]: Unused, can delete

File: src/main/java/com/sampleweb/pages/HomeKitchen.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    private String starReviewLocator = "//span[@class=\"a-icon-alt\"]";
+    private String primeLabelOrMoreLocator = "//*[@aria-label=\'Amazon Prime\' or @class=\'a-size-base a-color-secondary\']";
+    private String primeCheckBoxLocator = "//input[@name=\"s-ref-checkbox-419158031\"]";
+    private String sortByLocator = "//*[@id=\"s-result-sort-select\"]";
```
Comment (lewisbirks) [inline src/main/java/com/sampleweb/pages/HomeKitchen.java:25]: Unused, can delete

File: src/main/java/com/sampleweb/pages/HomeKitchen.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    public boolean isTitleCorrect(String title) throws Exception {
+        return verifyTitle(driver.getTitle(), title);
+    }
+
+    @FindBy(xpath = "//*[@id=\"leftNav\"]/ul[9]/div/li[1]/span/a")
+    protected WebElement averageCustomerReviewLinkLocator;
+
+    @FindBy(xpath = "")
+    protected WebElement sortBy;
```
Comment (lewisbirks) [inline src/main/java/com/sampleweb/pages/HomeKitchen.java:45]: Unused, can delete

File: src/main/java/com/sampleweb/pages/HomeKitchen.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+    public HomeKitchen directToKitchen() {
+        directToCategory("search-alias=kitchen");
+        return this;
+    }
+
+        public boolean isGlobalStoreSectionThere() {
```
Comment (lewisbirks) [inline src/main/java/com/sampleweb/pages/HomeKitchen.java:52]: This method is indented differently compared to the rest.

File: src/main/java/com/sampleweb/pages/HomeKitchen.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+            if (!currWebElement.findElement(By.xpath(primeLabelOrMoreLocator)).isDisplayed()){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public boolean isSubMenuCorrect(){
+        String menuText = driver.findElement(By.xpath(secondMenuLocator)).getText();
+        if (menuText.equals(secondMenuTitle)) {
```
Comment (lewisbirks) [inline src/main/java/com/sampleweb/pages/HomeKitchen.java:85]: Line 85 - 88 can be simplified to `return menuText.equals(secondMenuTitle);`

File: src/main/java/com/sampleweb/pages/HomeKitchen.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    public void goToSortHighReviews(){
+        averageCustomerReviewLinkLocator.click();
+        driver.findElement(By.xpath("//select/option[@value= \"review-rank\"]")).click();
+    }
+
+    public boolean checkSort(){
+        goToSortHighReviews();
+        boolean check = false;
```
Comment (lewisbirks) [inline src/main/java/com/sampleweb/pages/HomeKitchen.java:98]: Unused, can remove

File: src/main/java/com/sampleweb/pages/HomeKitchen.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+            } catch (org.openqa.selenium.NoSuchElementException e) {
+                return false;
+            }
+        }
+
+    public boolean isAverageCustomerReviewThere(){
+        try {
+            driver.findElement(By.xpath(averageCustomerReviewLocator));
+            return true;
+        } catch (org.openqa.selenium.NoSuchElementException e) {
```
Comment (lewisbirks) [inline src/main/java/com/sampleweb/pages/HomeKitchen.java:66]: Don't need the fully qualified name. Just import the `NoSuchElementException`.

File: src/main/java/com/sampleweb/pages/HomeKitchen.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+        directToCategory("search-alias=kitchen");
+        return this;
+    }
+
+        public boolean isGlobalStoreSectionThere() {
+            try {
+                driver.findElement(By.xpath(globalStoreBoxLocator));
+                driver.findElement(By.xpath(globalStoreLabelLocator));
+                return true;
+            } catch (org.openqa.selenium.NoSuchElementException e) {
```
Comment (lewisbirks) [inline src/main/java/com/sampleweb/pages/HomeKitchen.java:57]: Don't need the fully qualified name. Just import the `NoSuchElementException`.

File: src/main/java/com/sampleweb/pages/HomePage.java [hunk 10]
```diff
@@ -25,4 +25,18 @@ public WebElement getSearchField() {
     public void clickSubmitButton() {
         submitButtonLocator.click();
     }
+
+    public void emptySearch(){
+        getSearchField().sendKeys(" ");
```
Comment (lewisbirks) [inline src/main/java/com/sampleweb/pages/HomePage.java:30]: could maybe use `.clear()` instead of `.sendKeys(" ")`

**Review by lewisbirks — CHANGES_REQUESTED** [review]
Seems good

**Pull request closed without merging**